src/actions/definitionActions.js

```diff
@@ -99,21 +98,18 @@ export function browseDefinitionsAction(token, query, name) {
     const actions = asyncActions(name)
     dispatch(actions.start())
     try {
-      dispatch(uiBrowseUpdateList({ startQuery: true }))
       const result = await searchDefinitions(token, query)
-      const definitions = result.data
-      dispatch(actions.success({ add: definitions }))
-      const toAdd = map(definitions, component => EntitySpec.validateAndCreate(component.coordinates)).filter(e => e)
-      if (query.continuationToken) dispatch(uiBrowseUpdateList({ addAll: toAdd, data: result.continuationToken }))
-      else dispatch(uiBrowseUpdateList({ updateAll: toAdd, data: result.continuationToken }))
+      const definitionBodies = result.data.reduce((result, definition) => {
+        const coordinates = EntitySpec.validateAndCreate(definition.coordinates)
+        if (coordinates) result[coordinates] = definition
+        return result
+      }, {})
+      const allCoordinates = Object.keys(definitionBodies).map(EntitySpec.fromPath)
+      if (query.continuationToken) dispatch(actions.success({ addAll: allCoordinates, data: result.continuationToken }))
+      else dispatch(actions.success({ updateAll: allCoordinates, data: result.continuationToken }))
+
       const definitionActions = asyncActions(DEFINITION_BODIES)
-      definitions.forEach(component => {
```

Comment:
Removing this increase the performance, but adds some problem while operating directly on a specific definition.
If you check the deploy, and try to change something on a definition, you can't actually revert that change using the revert function.